Sources/PickBetter/SegmentedBetterPickerStyle.swift

```diff
@@ -0,0 +1,226 @@
+// SegmentedBetterPickerStyle.swift
+// PickBetter
+//
+// Copyright © 2024 MFB Technologies, Inc. All rights reserved. All rights reserved.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+import Foundation
+import SwiftUI
+
+/// 'Segmented' style that mimics Apple's PickerStyle.segmented
+public struct SegmentedBetterPickerStyle: BetterPickerStyle {
+    /// Width of the entire Picker. Required property in order to calculate cell size correctly.
+    private let frameWidth: CGFloat
+    /// Height of the entire Picker. Defaults to '32', the same height used in Apple's segmented Picker.
+    private let frameHeight: CGFloat
+    /// Horizontal alignment for the inner ``View`` of the cells. Defaults to ``.center`` alignment.
+    private let horizontalCellAlignment: HorizontalAlignment
+
+    /// Static value for the amount of padding between cell selection frame and parent frame
+    private let cellSelectionPadding: CGFloat = 2.0
+
+    /// Memberwise initializer
+    public init(
+        frameWidth: CGFloat,
+        frameHeight: CGFloat = 32,
+        horizontalCellAlignment: HorizontalAlignment = .center
+    ) {
+        self.frameWidth = frameWidth
+        self.frameHeight = frameHeight
+        self.horizontalCellAlignment = horizontalCellAlignment
+    }
+
+    /// Builds the overall picker ``View``.
+    public func makeView(_ configuration: Configuration) -> some View {
+        /// Full width of the `View` for a cell.
+        let cellWidth = (frameWidth / CGFloat(configuration.cellCount)) - 1.5
+        /// Index of the selected cell.
+        let selectCellIndex = configuration.selectionIndexSet.first
+
+        ZStack(alignment: .leading) {
+            if let selectCellIndex = selectCellIndex {
+                /// Rounded background to display selected item
+                RoundedRectangle(cornerRadius: 6)
+                    .fill(.white)
+                    .frame(width: cellWidth, height: frameHeight - (cellSelectionPadding * 2))
+                    .offset(x: CGFloat(selectCellIndex) * cellWidth)
+                    .animation(.easeInOut(duration: 0.25), value: selectCellIndex)
+                    .padding(cellSelectionPadding)
+                    .shadow(color: .secondary.opacity(0.3), radius: 2, y: 1)
+            }
+
+            addCellDividers(
```

Comment:
The dividers don't animate out like in the SwiftUI native style. It looks more pronounced on the bigger custom cell style. I'm fine with leaving that for a future PR if you are. If that's the route you want to go, do you mind creating an issue for it?

Reply:
Yeah, I created an issue. I think it's definitely usable without that.

https://github.com/MFB-Technologies-Inc/swiftui-pick-better/issues/14